extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/predictor/OnnxModelRunner.java

```diff
@@ -0,0 +1,26 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor;
+
+import ai.onnxruntime.OnnxTensor;
+import ai.onnxruntime.OrtEnvironment;
+import ai.onnxruntime.OrtException;
+import ai.onnxruntime.OrtSession;
+
+import java.util.Collections;
+
+public class OnnxModelRunner {
+
+    private OrtSession session;
+
+    private OrtEnvironment environment;
+
+    public OnnxModelRunner(byte[] onnxModelBytes) throws OrtException {
+        environment = OrtEnvironment.getEnvironment();
+        final OrtSession.SessionOptions options = new OrtSession.SessionOptions();
+        session = environment.createSession(onnxModelBytes, options);
+    }
+
+    public OrtSession.Result runModel(String[][] throttlingInferenceRow) throws OrtException {
+        final OnnxTensor inputTensor = OnnxTensor.createTensor(OrtEnvironment.getEnvironment(), throttlingInferenceRow);
+        return session.run(Collections.singletonMap("input", inputTensor));
+    }
```

Comment:
We'll need the environment anyway, so let's load it with the class instead of loading it when lots of requests try to do it and get blocked.
```
public class OnnxModelRunner {

    private static final OrtEnvironment ENVIRONMENT = OrtEnvironment.getEnvironment();
    
    private final OrtSession session;

    public OnnxModelRunner(byte[] onnxModelBytes) throws OrtException {
        session = ENVIRONMENT.createSession(onnxModelBytes, new OrtSession.SessionOptions());
    }

    public OrtSession.Result runModel(String[][] throttlingInferenceRow) throws OrtException {
        final OnnxTensor inputTensor = OnnxTensor.createTensor(ENVIRONMENT, throttlingInferenceRow);
        return session.run(Collections.singletonMap("input", inputTensor));
    }
```

Reply:
And i have a question, is the model implementation (session) immutable? Then we can reuse single instance.

Reply:
Yes after the model is fetched the `OrtSession session` is immutable.  When the cache is expired and `loadModelRunner` is triggered and we create new `OnnxModelRunner` instance with new session inside of it